airflow/serialization/serialized_objects.py

```diff
@@ -538,9 +538,9 @@ def serialize(
         elif isinstance(var, Resources):
             return var.to_dict()
         elif isinstance(var, MappedOperator):
-            return SerializedBaseOperator.serialize_mapped_operator(var)
+            return cls._encode(SerializedBaseOperator.serialize_mapped_operator(var), type_=DAT.OP)
```

Comment:
@potiuk @uranusjr this PR needs a closer look.  what i'm doing is making it so that when we serialize an Operator, we actually run it through `cls._encode` so that on the other side we actually know what the object is.  we got away with *not* doing that up to now because `task` is always serialized as part of something else (e.g. TI or DAG) and custom serialization handles it.  but with AIP-44 there are times when we want to be able to serialize a task object directly and thus we need a way to do it. see below for backcompat code for deserializing old and new way.

Reply:
It's fine for me. If we want to continue using airflow serds for all pydantic objects as well, this is the right thing to do. I **think** we don't - we should be able to rely purely on Pydantic serializing TaskInstance as TaskInstancePydantic. But yes - if we want to do it - it's fine.

Reply:
I don't know what's at stake here? What's the alternative way?  If that' better, why not do it?